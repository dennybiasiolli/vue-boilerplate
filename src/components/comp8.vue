<template>
  <div>
    <p>
      Coumponent mounted {{ count }} times
    </p>
    <p>
      Counter is even? {{ isCountEven }}
    </p>
    <p>
      Counter is odd? {{ isCountOdd }}
    </p>
    <p>
      State with local data: {{ countPlusLocalData }}
    </p>
    <p>
      State with local computed data: {{ countPlusLocalComputed }}
    </p>
    <p>
      Local state: {{ localState }}
    </p>
  </div>
</template>

<script>
import { createNamespacedHelpers } from 'vuex'
const { mapState, mapGetters, mapActions } = createNamespacedHelpers('a')

export default {
  data() {
    return {
      localCount: 10,
      localState: 1000
    }
  },
  computed: Object.assign(
    // local computed properties
    {
      localComputed() {
        return 100
      }
    },

    // state properties, different use cases
    mapState({
      // arrow functions can make the code very succinct!
      count: state => state.count,

      // passing the string value 'count' is same as `state => state.count`
      countAlias: 'count',

      // to access local state with `this`, a normal function must be used
      countPlusLocalData(state) {
        return state.count + this.localCount
      },
      countPlusLocalComputed(state) {
        return state.count + this.localComputed
      }
    }),

    // getter properties (like computed properties for store)
    mapGetters(['isCountEven', 'isCountOdd'])
  ),
  mounted() {
    this.incrementAsync().then(() => {})
    this.localState++
  },
  methods: Object.assign({}, mapActions(['incrementAsync']))
}
</script>
