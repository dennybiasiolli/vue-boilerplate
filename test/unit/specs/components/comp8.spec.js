import Vuex from 'vuex'
import { shallow, createLocalVue } from 'vue-test-utils'
import comp8 from '@/components/comp8'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('comp8', () => {
  let component, vm, element
  let state, mutations, getters

  beforeEach(() => {
    state = {
      count: 0
    }

    mutations = {
      increment: jest.fn(state => state.count++)
    }

    getters = {
      isCountEven: jest.fn().mockReturnValue(false),
      isCountOdd: jest.fn().mockReturnValue(true)
    }

    const store = new Vuex.Store({
      state,
      mutations,
      getters
    })

    component = shallow(comp8, { store, localVue })
    vm = component.vm
    element = component.element
  })

  test('should match snapshot', () => {
    expect(element).toMatchSnapshot()
  })

  // Evaluate the results of functions in
  // the raw component options
  test('sets the correct default data', () => {
    expect(typeof comp8.data).toBe('function')
    const defaultData = comp8.data()
    expect(defaultData.localCount).toBe(10)
    expect(defaultData.localState).toBe(1000)
    expect(comp8.computed.localComputed()).toBe(100)
    expect(typeof comp8.mounted).toBe('function')
    expect(typeof comp8.methods.increment).toBe('function')
  })

  // Inspect the component instance on mount
  test('correctly sets the data when created', () => {
    expect(vm.localCount).toBe(10)
    expect(vm.localState).toBe(1001)
    expect(vm.localComputed).toBe(100)
    expect(vm.count).toBe(1)
    expect(vm.countAlias).toBe(1)
    expect(vm.countPlusLocalData).toBe(11)
    expect(vm.countPlusLocalComputed).toBe(101)
    expect(vm.isCountEven).toBe(false)
    expect(vm.isCountOdd).toBe(true)
    expect(mutations.increment).toHaveBeenCalled()
    expect(mutations.increment.mock.calls[0][1]).toBeUndefined()
    expect(getters.isCountEven).toHaveBeenCalled()
    expect(getters.isCountOdd).toHaveBeenCalled()
  })
})
