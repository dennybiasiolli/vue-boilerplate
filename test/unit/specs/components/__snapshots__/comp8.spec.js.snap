// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comp8 should match snapshot 1`] = `
<div>
  <p>
    
    Coumponent mounted 0 times
  
  </p>
   
  <p>
    
    Counter is even? false
  
  </p>
   
  <p>
    
    Counter is odd? true
  
  </p>
   
  <p>
    
    State with local data: 10
  
  </p>
   
  <p>
    
    State with local computed data: 100
  
  </p>
   
  <p>
    
    Local state: 1001
  
  </p>
</div>
`;
